### PR TITLE
feat(formatter): unquote valid members

### PR DESCRIPTION
**Pull request by pradel**
## Summary

Hey all (first-time contributor to rome), to get a bit more familiar with the codebase I decided to try to fix issue https://github.com/rome/tools/issues/982.

A lot of changes in this pr are related to this new formatting applied to the codebase. The important files are:

For for property keys:
- `internal/formatter/builders/js/objects/JSStaticPropertyKey.ts` changes for property keys.
- `internal/formatter/test-fixtures/objects/property-key/input.test.md` test output.

For computed members:
- `internal/formatter/builders/js/expressions/JSMemberExpression.ts` changes for computed members.
- `internal/formatter/test-fixtures/objects/computed-member/input.js` test output.

From the snapshots, this is how the change looks like:

### Property keys `Input`

```js
const foo = {
  "bar": true,
  "foo-bar": true,
};
```

### Property keys `Output`

```js
const foo = {
	bar: true,
	"foo-bar": true,
};
```

### Computed members `Input`

```js
const foo = {};

foo["bar"] = true;
foo["foo-bar"] = true;
```

### Computed members `Output`

```js
const foo = {};

foo.bar = true;
foo["foo-bar"] = true;
```

## Test Plan

Tests are passing properly ✅ 

```
➜ ./rome check --apply
 Building trunk 
! This command has been hidden. Consider its usage to be experimental and do not expect support or
ℹ 1_734 files linted
✔ No known problems!
```

```
➜ ./rome test 
 Building trunk 
! This command has been hidden. Consider its usage to be experimental and do not expect support or
✔ 3_844 tests passed!
```

### CODE REVIEW

**Comment by ematipico** [comment]
Hi @pradel , thank you for looking into this!

Regarding your issue, here's the steps you can follow to understand where to look:
- create a dummy file e.g. `test.js`
- inside the file you can write:

```js
a["b"] = "bew";
a.b = "bew"	
```

which are the two cases that you want to analyze (format from the first case to the second case)

- run the command `./rome parse test.ts`

The output should contain something like this:

```js
JSExpressionStatement {
  expression: JSAssignmentExpression {
    operator: "="
    right: JSStringLiteral {value: "bew"}
    left: JSMemberExpression {
      object: JSReferenceIdentifier {name: "a"}
      property: JSComputedMemberProperty {value: JSStringLiteral {value: "b"}}
    }
  }
}
JSExpressionStatement {
  expression: JSAssignmentExpression {
    operator: "="
    right: JSStringLiteral {value: "bew"}
    left: JSMemberExpression {
      object: JSReferenceIdentifier {name: "a"}
      property: JSStaticMemberProperty {value: JSIdentifier {name: "b"}}
    }
  }
}

```

From here, you can understand that we want to apply a different formatting inside the `JSMemberExpression` node. So you can open the file `JSMemberExpression.ts` insider the formatter package.

As we want create new nodes (and replace with the existing ones), you can use the function `jsStaticMemberProperty` that you can export from the file `ast/js/expressions/JSStaticMemberProperty.ts`.

I hope it helps.

**Comment by pradel** [comment]
@ematipico thanks for the help I managed to make it work 🎉 
I updated the description of the pull request to point to the important parts of the changes. Let me know if something is missing or wrong :)

File: internal/formatter/test-fixtures/objects/computed-member/input.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+const foo = {};
```
Comment (ematipico) [inline internal/formatter/test-fixtures/objects/computed-member/input.js:1]: I might make sense to add some mixed/more cases, e.g.:

```js
foo.bar["bar"]["lorem_ispsum"]["lorem-ipsum"]
```

And so on. Let's try to cover some edge case.

**Review by ematipico — CHANGES_REQUESTED** [review]
It looks good to me! I would add some more test case inside the tests

File: internal/formatter/test-fixtures/objects/computed-member/input.js (hunk 1 above)
Comment (pradel) [reply internal/formatter/test-fixtures/objects/computed-member/input.js:1]: Added some edge cases 👍

**Review by ematipico — APPROVED** [review]
